### PR TITLE
Add docs on DAG & IngressRoute objects

### PR DESCRIPTION
**Pull request by stevesloka**
Updates #644 

Signed-off-by: Steve Sloka <steves@heptio.com>

### CODE REVIEW

File: docs/about.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # About Contour and Envoy
 
-At the highest level Contour is a proxy that fulfills the Envoy Cluster Discovery Service (CDS) API by translating Kubernetes Ingress objects into JSON configuration fragments.
+At the highest level Contour is a proxy that fulfills the Envoy Cluster Discovery Service (CDS) API by translating Kubernetes Ingress objects and IngressRoute objects into JSON configuration fragments.
```
Comment (alexbrand) [inline docs/about.md]: Now that we no longer support the JSON API, I think we can update "JSON configuration fragments" as well. Maybe something like: "into Envoy configuration objects" ?

File: docs/about.md [hunk 2]
```diff
@@ -11,10 +11,18 @@ Unlike other ingress controllers, Envoy supports continual reconfiguration witho
 
 ## How Contour works
 
+Contour supports both Ingress objects and IngressRoute objects which are ways to define traffic routes into a cluster. 
+They are different in how the objects are structured as well as how they are implemented, however, are identical at their core intent, to simply route traffic.
```
Comment (alexbrand) [inline docs/about.md]: nit: the intent of the Ingress and IngressRoute objects is to configure the routing of ingress traffic, not to actually route traffic.

File: docs/about.md (hunk 1 above)
Comment (stevesloka) [reply docs/about.md]: :+1:

File: docs/about.md (hunk 2 above)
Comment (stevesloka) [reply docs/about.md]: :+1:

**Pull request merged**